doc/cfg_reload_xact_enhance/Config_Reload_Enhancement_Xact_HLD.md

```diff
@@ -0,0 +1,230 @@
+# Config Reload Enhancement - Introduce the Transaction Mechanism
+# High Level Design Document
+## Rev 0.1
+
+## Table of Content
+- [Table of Content](#table-of-content)
+- [Revision](#revision)
+- [Scope](#scope)
+- [Definitions/Abbreviations](#definitions-abbreviations)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [Architecture Design](#architecture-design)
+- [High-Level Design](#high-level-design)
+  * [The Temporarily Files Generated by This Enhancement](#the-temporarily-files-generated-by-this-enhancement)
+  * [New Error Messages by This Enhancement](#new-error-messages-by-this-enhancement)
+  * [CLI/YANG model Enhancements](#cli-yang-model-enhancements)
+  * [Config DB Enhancements](#config-db-enhancements)
+- [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+- [Restrictions/Limitations](#restrictions-limitations)
+- [Testing Requirements/Design](#testing-requirements-design)
+  * [Unit Test Cases](#unit-test-cases)
+  * [System Test Cases](#system-test-cases)
+- [Open/Action Items](#open-action-items)
+
+### Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 2022/3/18   |     Charlie Chen   | Initial version                   |
+
+### Scope
+
+This document describes the enhancement based on the existing design of the
+Sonic CLI command 'config reload'. The enhancement introduces the transaction
+mechanism to guarantee that only one 'config reload' session runs in the system
+at the same time. The design regarding the determination of a transaction done
```

Comment:
Please verify the existing behavior. Currently config reload cannot run in parallel unless force option is used.